agent/consul/gateways/controller_gateways.go

```diff
@@ -840,151 +828,132 @@ func newGatewayMeta(gateway *structs.APIGatewayConfigEntry, bound structs.Config
 	}).initialize()
 }
 
-// gatewayConditionGenerator is a simple struct used for isolating
-// the status conditions that we generate for our components
-type gatewayConditionGenerator struct {
-	now *time.Time
-}
-
-// newGatewayConditionGenerator initializes a status conditions generator
-func newGatewayConditionGenerator() *gatewayConditionGenerator {
-	return &gatewayConditionGenerator{
-		now: pointerTo(time.Now().UTC()),
-	}
+// gatewayAccepted marks the APIGateway as valid.
+func gatewayAccepted() structs.Condition {
+	return structs.NewGatewayCondition(
+		structs.GatewayConditionAccepted,
+		structs.ConditionStatusTrue,
+		structs.GatewayReasonAccepted,
+		"gateway is valid",
+		structs.ResourceReference{},
+	)
 }
 
 // invalidCertificate returns a condition used when a gateway references a
 // certificate that does not exist. It takes a ref used to scope the condition
 // to a given APIGateway listener.
-func (g *gatewayConditionGenerator) invalidCertificate(ref structs.ResourceReference, err error) structs.Condition {
-	return structs.Condition{
-		Type:               "Accepted",
-		Status:             "False",
-		Reason:             "InvalidCertificate",
-		Message:            err.Error(),
-		Resource:           pointerTo(ref),
-		LastTransitionTime: g.now,
-	}
+func invalidCertificate(ref structs.ResourceReference, err error) structs.Condition {
+	return structs.NewGatewayCondition(
+		structs.GatewayConditionResolvedRefs,
+		structs.ConditionStatusFalse,
+		structs.GatewayListenerReasonInvalidCertificateRef,
+		err.Error(),
+		ref,
+	)
 }
 
 // invalidCertificates is used to set the overall condition of the APIGateway
 // to invalid due to missing certificates that it references.
-func (g *gatewayConditionGenerator) invalidCertificates() structs.Condition {
-	return structs.Condition{
-		Type:               "Accepted",
-		Status:             "False",
-		Reason:             "InvalidCertificates",
-		Message:            "gateway references invalid certificates",
-		LastTransitionTime: g.now,
-	}
+func invalidCertificates() structs.Condition {
+	return structs.NewGatewayCondition(
+		structs.GatewayConditionResolvedRefs,
+		structs.ConditionStatusFalse,
+		structs.GatewayListenerReasonInvalidCertificateRef,
+		"gateway references invalid certificates",
+		structs.ResourceReference{},
+	)
 }
 
-// gatewayAccepted marks the APIGateway as valid.
-func (g *gatewayConditionGenerator) gatewayAccepted() structs.Condition {
-	return structs.Condition{
-		Type:               "Accepted",
-		Status:             "True",
-		Reason:             "Accepted",
-		Message:            "gateway is valid",
-		LastTransitionTime: g.now,
-	}
+// gatewayListenerNoConflicts marks an APIGateway listener as having no conflicts within its
+// bound routes
+func gatewayListenerNoConflicts(ref structs.ResourceReference) structs.Condition {
+	return structs.NewGatewayCondition(
+		structs.GatewayConditionListenersConfigured,
```

Comment:
I changed out our custom "Conflicted" type for "ListenersConfigured" for two reasons:
1. to be closer to the spec
2. It was the only condition where `ConditionStatusFalse` meant there were no issues and that just seemed a bit counter intuitive

Reply:
I missed that one when I looked at the spec! I'll add that in and move it back to the condition being conflicted and use a custom `RouteConflict` as the reason